server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/InternalComposite.java

```diff
@@ -447,5 +449,63 @@ public int size() {
                 }
             };
         }
+
+        private <T> int compareInternal(T a, T b, Comparator<T> comparator) {
+            if (a == null && b == null) {
+                return 0;
+            }
+            if (a == null) {
+                return -1;
+            }
+            if (b == null) {
+                return 1;
+            }
+            return comparator.compare(a,b);
+        }
+
+        /**
+         * @param a - first object to compare
+         * @param b - second object to compare
+         * @return compares two Objects of unknown type
+         * @throws IllegalStateException if they a and b not of the same class or their class doe's not impl java.lang.Comparable
+         */
+        private static int objectCompare(Object a, Object b) {
+            if (a.getClass() != b.getClass()) {
+                throw new IllegalStateException("expecting values to be of the same type but got: " + a
+                    + "(" + a.getClass() +"), " + b + "(" + b.getClass() +")");
+            }
+            if (Comparable.class.isAssignableFrom(a.getClass())) {
+                Comparable<Object> ca = (Comparable<Object>) a;
+                return ca.compareTo(b);
+            }
+            throw new IllegalStateException("expecting values types to implement java.lang.Comparable but got: " + a.getClass());
+        }
+
+        @Override
+        public int compareTo(Map<String,Object> that) {
+            if (that == this) {
+                return 0;
+            }
+            Iterator<Entry<String, Object>> thisIterator = this.entrySet().iterator();
```

Comment:
The iterator returns the formatted values so you should simply iterate directly on the `values` array ?